posts/2024-03-12-argocd-drift-pt1.adoc

```diff
@@ -0,0 +1,161 @@
+
+
+GitOps is more than just a set of tools or practices. It's a cultural shift in how infrastructure and applications are managed, drawing from proven development practices and extending them to operations. By tightly integrating Git at the center of the operational framework, GitOps ensures reliability, transparency, and efficiency in modern deployment pipelines. In essence, GitOps ensures that the actual state of a cluster is always equal to the desired state in Git.
+
+After understanding the principles of GitOps, it becomes evident that achieving its full potential requires tools designed with these principles in mind. Let's explore what tools we can use.
+
+== How does ArgoCD help enable GitOps? == 
+
+ArgoCD is a declarative, GitOps continuous delivery tool for Kubernetes. It works by ensuring that a Kubernetes cluster's state matches the configurations defined in a Git repository. It facilitates the automatic deployment and synchronization of applications to the desired environments.  
+
+We chose to use ArgoCD with Open Liberty because while typical Continuous Deployment (CD) systems adhere to the first two GitOps principles (using declarative configurations and maintaining a versioned, immutable source of truth), they often fall short when trying to implemet the latter two principles (ensuring automated delivery and maintaining closed-loop feedback). ArgoCD, however, does help with these two GitOps principles as well.
+
+How does Argo CD help with these two GitOps principles?
+
+* Automated Delivery: ArgoCD continuously and automatically syncs applications when changes are committed to the Git repository. This ensures deployments are consistent with the repository, bridging the gap between development commits and operational deployments. 
+
+* Closed-Loop Feedback & Control: ArgoCD's real-time monitoring of application performance ensures that the live state is always aligned with the Git repository's state. If a discrepancy is detected, ArgoCD offers visual representations of the divergence and provides mechanisms to reconcile those differences, ensuring constant alignment with the intended state. 
+
+Integrating this into the application we explored at the start of this blog, you'll see ArgoCD added into our pipeline, ensuring that our cluster remains in sync with what is defined in our Git repository.
+
+[.img_border_light]
+image::/img/blog/argocd4.png[,width=70%,align="center"]
+
+== How can we make use of GitOps and ArgoCD? == 
+
+If you’d like to see how you can practically use GitOps principles and ArgoCD to deploy Open Liberty applications, we've provided a detailed tutorial style blog which follows on from this introductory blog. This follow-on blog takes you through, set-by-step, how you can take an existing Open Liberty application and introduce ArgoCD to help solve configuration drift in your own applications.  
+
+
+== Summary and Next Steps == 
```

Comment:
this section doesn't add much, I'd consider cutting it as the previous section sums things up and provides a call to action